docker/importer/volume/surveillance-mediator/endpoint-resource-resolver-dhis.json

```
@@ -0,0 +1,184 @@
+            "attribute": "Rv8WM2mTuS5",
+            "value": 0
+          },
+          "1074":{
+            "attribute": "Rv8WM2mTuS5",
+            "value": 0
+          }
+        }
+      }
+    },
+    "requestBody.data.observations[1].value.valueCodeableConcept.coding[0].code": {
+      "key": "attributes[]+",
+      "transform": {
+        "function": "mapCodes",
+        "parameters": {
+          "male": {
+            "attribute": "oindugucx72",
+            "value": "Male"
+          },
+          "female": {
+            "attribute": "oindugucx72",
+            "value": "Female"
+          }
+        }
+      }
+    },
+    "transforms.patinfo_ID": "attributes[]+",
+
+    "transforms.incidentDate": {
+      "key": "incidentDate",
+      "transform": {
+        "function": "dateTimeToDate"
+      }
+    },
+
+    "requestBody.data.observations[3].value.valueCodeableConcept.coding[0].code": {
+      "key": "event.dataValues[]+",
+      "transform": {
+        "function": "mapCodes",
+        "parameters": {
+          "Y": {
+            "dataElement": "JGnHr6WI3AY",
+            "value": "Yes"
+          },
+          "N": {
+            "dataElement": "JGnHr6WI3AY",
+            "value": "No"
+          },
+          "U": {
+            "dataElement": "JGnHr6WI3AY",
+            "value": "Unknown"
+          }
+        }
+      }
+    },
+    "requestBody.data.observations[5].value.valueCodeableConcept.coding[0].code": {
+      "key": "event.dataValues[]+",
+      "transform": {
+        "function": "mapCodes",
+        "parameters": {
+          "Y": {
+            "dataElement": "sJeIFfhX8BE",
+            "value": "Yes"
+          },
+          "N": {
+            "dataElement": "sJeIFfhX8BE",
+            "value": "No"
+          },
+          "U": {
+            "dataElement": "sJeIFfhX8BE",
+            "value": "Unknown"
+          }
+        }
+      }
+    },
+    "requestBody.data.observations[6].value.valueCodeableConcept.coding[0].code": {
+      "key": "event.dataValues[]+",
+      "transform": {
+        "function": "mapCodes",
+        "parameters": {
+          "Y": {
+            "dataElement": "TzqawmlPkI5",
+            "value": "Yes"
+          },
+          "N": {
+            "dataElement": "TzqawmlPkI5",
+            "value": "No"
+          },
+          "U": {
+            "dataElement": "TzqawmlPkI5",
+            "value": "Unknown"
+          }
+        }
+      }
+    },
+    "transforms.patcourse_dateonset": "event.dataValues[]+"
+  },
+  "inputTransforms": {
+    "patinfo_ID": "{ 'attribute': constants.attributes.patinfo_ID, 'value': requestBody.data.identifier }",
+    "patinfo_ageonset": "{ 'attribute': constants.attributes.patinfo_ageonset, 'value': requestBody.data.observations[0].value.valueQuantity.value }",
+
+    "incidentDate": "$now()",
+
+    "patcourse_dateonset": "{ 'dataElement': constants.dataElements.patcourse_dateonset, 'value': requestBody.data.observations[4].value.valueDateTime }"
```

@bradsawadye Something like this has no code duplication

```suggestion
    "patinfo_sex": "{ 'attribute': constants.attributes.patinfo_sex, 'value': $lookup(constants.sexMap, requestBody.data.observations[1].value.valueCodeableConcept.coding[0].code) }",

    "incidentDate": "$now()",
    "patcourse_asymp": "{ 'dataElement': constants.dataElements.patcourse_asymp, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[3].value.valueCodeableConcept.coding[0].code) }",
    "Comcond_any": "{ 'dataElement': constants.dataElements.Comcond_any, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[5].value.valueCodeableConcept.coding[0].code) }",
    "patcourse_dateonset": "{ 'dataElement': constants.dataElements.patcourse_dateonset, 'value': requestBody.data.observations[4].value.valueDateTime }",
    "expo_travel": "{ 'dataElement': constants.dataElements.expo_travel, 'value': $lookup(constants.yesNoUnknownMap, requestBody.data.observations[6].value.valueCodeableConcept.coding[0].code) }"
```